src/shared/cloudformation/templateRegistry.ts

```diff
@@ -91,6 +100,7 @@ export class CloudFormationTemplateRegistry {
      */
     public removeTemplateFromRegistry(templateUri: vscode.Uri): void {
         const pathAsString = pathutils.normalize(templateUri.fsPath)
+        this.assertAbsolute(pathAsString)
```

Comment:
Do we care about this failing on the deletion? The map doesn't care.

Reply:
similar use-case as a type system: asserts declare invariants/correctness. If a relative path is given to `removeTemplateFromRegistry` then that indicates a bug, so failing early may be helpful.

But your point about `Uri` makes sense.

Reply:
![](https://i.imgur.com/PwaNKrK.gif)